File: src/main/java/com/ceres/blip/models/database/VehicleModel.java
package com.ceres.blip.models.database;

import com.ceres.blip.models.jpa_helpers.enums.VehicleTypes;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "vehicle")
public class VehicleModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "registration_number", nullable = false, length = 20)
    private String registrationNumber;

    @Column(name = "capacity", nullable = false)
    private Integer capacity;

//    @Enumerated(EnumType.STRING)
//    @Column(name = "category", nullable = false, length = 50)
//    private VehicleTypes category;

    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false, length = 20)
    private VehicleTypes type;

    @ColumnDefault("'ACTIVE'")
    @Column(name = "status", nullable = false, length = 20)
    private String status;

    @Basic
    @Column(name = "partner_code")
    private String partnerCode;

    @ColumnDefault("CURRENT_TIMESTAMP")
    @Column(name = "created_at", nullable = false)
    private Timestamp createdAt;

    @Column(name = "created_by", nullable = false)
    private Long createdBy;

}

File: frontend/src/app/components/models/vehicle.model.ts
export interface Vehicle {
    id?: number;
    registration_number: string;
    capacity: number;
    type: VehicleType;
    status?: string;
    partner_code?: string;
    created_at?: string;
    created_by?: number;
    vehicle_category?: string;
}

export enum VehicleType {
    BUS = 'BUS',
    DRONE = 'DRONE',
    MINI_VAN = 'MINI_VAN',
    MINI_BUS = 'MINI_BUS',
    TRUCK = 'TRUCK'
}
